src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/Http/ServerCertificateValidationProvider.cs

```diff
@@ -0,0 +1,156 @@
+﻿// <copyright file="ServerCertificateValidationProvider.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Resources.Http
+{
+    internal class ServerCertificateValidationProvider
+    {
+        private static readonly ServerCertificateValidationProvider InvalidProvider =
+            new ServerCertificateValidationProvider(null);
+
+        private readonly X509Certificate2Collection trustedCertificates;
+
+        private ServerCertificateValidationProvider(X509Certificate2Collection trustedCertificates)
+        {
+            if (trustedCertificates == null)
+            {
+                this.trustedCertificates = null;
+                this.ValidationCallback = null;
+                return;
+            }
+
+            this.trustedCertificates = trustedCertificates;
+            this.ValidationCallback = (sender, cert, chain, errors) =>
+                this.ValidateCertificate(new X509Certificate2(cert), chain, errors);
+        }
+
+        public RemoteCertificateValidationCallback ValidationCallback { get; }
+
+        public static ServerCertificateValidationProvider FromCertificateFile(string certificateFile)
+        {
+            if (!File.Exists(certificateFile))
+            {
+                AWSXRayEventSource.Log.FailedToValidateCertificate(nameof(ServerCertificateValidationProvider), "Certificate File does not exist");
+                return InvalidProvider;
+            }
+
+            var trustedCertificates = new X509Certificate2Collection();
+            if (!LoadCertificateToTrustedCollection(trustedCertificates, certificateFile))
+            {
+                AWSXRayEventSource.Log.FailedToValidateCertificate(nameof(ServerCertificateValidationProvider), "Failed to load certificate in trusted collection");
+                return InvalidProvider;
+            }
+
+            return new ServerCertificateValidationProvider(trustedCertificates);
+        }
+
+        private static bool LoadCertificateToTrustedCollection(X509Certificate2Collection collection, string certFileName)
+        {
+            try
+            {
+                collection.Import(certFileName, null, (X509KeyStorageFlags)32);
```

Comment:
otherwise if we have to use EphemeralKeySet flag I would suggest checking for the runtimes lower than net472 or lower than netstandard2.1 where X509KeyStorageFlags.EphemeralKeySet (32) doesn't exist. Otherwise we will eventually throw here. If we need I'd recommend doing this check much earlier so we can fallback gracefully

Reply:
Thanks for pointing this out. I have removed EphemeralKey.